monkey/infection_monkey/exploit/caching_agent_binary_repository.py

```diff
@@ -32,15 +34,10 @@ def get_agent_binary(
             return io.BytesIO(self._download_binary_from_island(operating_system))
 
     @lru_cache(maxsize=None)
-    def _download_binary_from_island(self, operating_system: OperatingSystem) -> bytes:
+    def _download_binary_from_island(self, operating_system: OperatingSystem) -> Optional[bytes]:
         os_name = operating_system.value
-
-        response = requests.get(  # noqa: DUO123
-            f"{self._island_url}/api/agent-binaries/{os_name}",
-            verify=False,
-            timeout=MEDIUM_REQUEST_TIMEOUT,
-        )
-
-        response.raise_for_status()
-
-        return response.content
+        try:
+            return self._island_api_client.get_agent_binary(os_name)
+        except Exception as exc:
+            logger.warning(f"Error connecting to control server: {exc}")
+            return None
```

Comment:
Don't hide this error. We can add an `RetrievalError` to `infection_monkey.cc.exploit.i_agent_binary_repository.py` and wrap the `IslandAPIErrors` in that to keep the interface simple. Be sure to update the docstring.